src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java

```diff
@@ -965,7 +968,7 @@ public Runnable getEditsRunnable(int version, Iterable<TextEdit> edits, String n
                     } else {
                         text = text.replace(DocumentUtils.WIN_SEPARATOR, DocumentUtils.LINUX_SEPARATOR);
                         if (end >= 0) {
-                            if (end - start <= 0 || end - start != text.length()) {
```

Comment:
@gayanper I guess `&&` should be used instead of the `ll`. Am I missing something?

Reply:
@NipunaRanasinghe this is to handle the following situations
- The first condition handles a true insertion where the start and end is 0 or if the start is greater than end
- The second condition is to handle the scenario where lookupstring and actual lookup object is different. That is the editText contains text which might have parenthesis and brackets, but the lookup string wont have that.

But i think i need to check on this one more time according to the new contract you mentioned in the comment on the issue. I will check this tonight and see if we can get rid of this second condition.

Reply:
@gayanper Yeah its better if we can validate whether we need this rule, as when testing your fix with BAL LS, few false insertions were occurred due to the new rule :/

Reply:
@NipunaRanasinghe i think in my latest commit i have removed the second condition :)

Reply:
@gayanper My Bad :( Didn't notice it